core/chain-configs/src/genesis_config.rs

```diff
@@ -293,18 +293,26 @@ impl<'de, F: FnMut(StateRecord)> Visitor<'de> for RecordsProcessor<&'_ mut F> {
     where
         A: MapAccess<'de>,
     {
+        let mut s = Some(self);
+        let mut has_records_field = false;
         while let Some(key) = map.next_key::<String>()? {
             match key.as_str() {
                 "records" => {
-                    map.next_value_seed(self)?;
-                    return Ok(());
+                    map.next_value_seed(
+                        s.take().ok_or_else(|| de::Error::custom("duplicate field: records"))?,
+                    )?;
+                    has_records_field = true;
```

Comment:
(sorry if the mesasge is duplicated, there was some glitch when submitting the comment)

nit: this is not as readable as it could be. The specific issue is that the inner expression, `s.take().ok_or_else(|| de::Error::custom("duplicate field: records"))?,`, has non-trivial control-flow (exiting the function on duplicates), and it is hidden in an outre function call.

It's better if control flow has a dediccated statemen. So, I'd written this as 

```
let s = s.take().ok_or_else(|| de::Error::custom("duplicate field: records"))?;
map.next_value_seed(s);
```

I also like using names `me` or `this` for self, but variable.